uvicorn/main.py

```diff
@@ -527,10 +527,11 @@ def run(self, sockets=None):
         logger.info("Shutting down")
 
         # Stop accepting new connections.
-        for socket in sockets or []:
-            socket.close()
         for server in self.servers:
             server.close()
+        for socket in sockets or []:
+            if socket != None:
```

Comment:
I don't think items in `sockets` can be `None` here, right?

Reply:
correct, just one step too protective